docs/gaslessFeature.md

```diff
@@ -0,0 +1,90 @@
+
+dapp frontend using ethers.js
+
+```ts
+      // define UserOp
+      const userOp: UserOperationStruct = {
+          callContract: realGameContract.address,
+          callData: realGameContractCallData,
+          callGasLimit: gasToExecuteRealGameContractCallData,
+          verificationGasLimit: gasToVerifyPaymaster,
+          maxFeePerGas: gasPrice,
+          maxPriorityFeePerGas: gasPrice,
+          paymasterAndData: PAYMASTER_CONTRACT_ADDRESS 
+      }
+      
+      // 1. construct and send gasless transaction via native sendTransaction
+      const abiCoder = new ethers.utils.AbiCoder();
+      const userOp = abiCoder.encode(["tuple(address callContract, bytes callData, uint256 callGasLimit, uint256 verificationGasLimit, uint256 maxFeePerGas, uint256 maxPriorityFeePerGas, bytes paymasterAndData) UserOperation"], [userOp]);
+      // first 4 bytes of keccak hash of handleOp((address,bytes,uint256,uint256,uint256,uint256,bytes))
+      const fnSelector = "fb4350d8";
+      // gasless payload = ENTRYPOINT_HANDLE_OP_SELECTOR + abiEncode(UserOperation)
+      const payload = "0x" + fnSelector + userOp.slice(2);
+      const gaslessTx = {
+        from: whitelistUser.address,
+        to: ENTRYPOINT_CONTRACT_ADDRESS,
+        data: payload,
+        gasPrice: 0,
+        gasLimit: 1000000,
+        value: 0,
+      }
+      const signer = new ethers.Wallet("private key");
+      const tx = await signer.sendTransaction(gaslessTx);
+      await tx.wait();
+      // 2. or just use ethers contract factory with EntryFactory and call directly it
+      {
+        // Send tx with a valid user.
+        const EntryPoint = await ethers.getContractFactory("EntryPoint");
+        const entrypoint = await EntryPoint.attach(ENTRYPOINT_CONTRACT_ADDRESS);
+        const tx = await entryPoint.connect(whitelistUser).handleOp(userOp, {gasLimit: 100000, gasPrice: 0});
+        await tx.wait();
+      }
+```
+
+### How to configure a paymaster contract
+
+From previous example we have deployed contract like [this one](https://github.com/godwokenrises/account-abstraction/blob/gw-gasless/contracts/samples/GaslessDemoPaymaster.sol). However this contract should be configurable in order to make it workable through web3 api. In order to do this, please execute the following smart contract methods:
+1. Execute `deposit` [method](https://github.com/godwokenrises/account-abstraction/blob/541f7cac9d83e75d152e7a58bec6d97b51221012/contracts/core/GaslessBasePaymaster.sol#L68) on `paymaster` in order to add balance. This is required as all gasless tx costs will be covered by these balance on `paymaster` contract.
+2. Execute `addStake` [method](https://github.com/godwokenrises/account-abstraction/blob/541f7cac9d83e75d152e7a58bec6d97b51221012/contracts/core/GaslessBasePaymaster.sol#L85) in order to register your `paymaster` on the `entrypoint`. This will add `paymaster` contract allowance to perform as `gasless provider` for you project.
```

Comment:
```suggestion
2. Execute `addStake` [method](https://github.com/godwokenrises/account-abstraction/blob/541f7cac9d83e75d152e7a58bec6d97b51221012/contracts/core/GaslessBasePaymaster.sol#L85) in order to register your `paymaster` on the `entrypoint`. This will add `paymaster` contract allowance to perform as a `gasless provider` for your project.
```